packages/orbit-components/src/Stack/__tests__/index.test.js

```diff
@@ -1,480 +1,163 @@
-    expect(stack).toHaveStyleRule("flex-grow", "1");
-    expect(stack).toHaveStyleRule("justify-content", "flex-start");
-    expect(stack).toHaveStyleRule("align-items", "flex-start");
-    expect(stack).toHaveStyleRule("align-content", "flex-start");
-    expect(stack).toHaveStyleRule("margin", "0 16px 0 0!important", {
-      modifier: "& > *",
-    });
+    expect(screen.getByTestId("test")).toHaveStyle({ flexShrink: 1 });
   });
-  it("should contain desktop styles", () => {
-    expect(stack).toHaveStyleRule("display", "inline-flex", {
-      media: getBreakpointWidth(QUERIES.DESKTOP, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("flex-wrap", "wrap", {
-      media: getBreakpointWidth(QUERIES.DESKTOP, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("flex-shrink", "1", {
-      media: getBreakpointWidth(QUERIES.DESKTOP, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("flex-grow", "0", {
-      media: getBreakpointWidth(QUERIES.DESKTOP, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("flex-basis", "auto", {
-      media: getBreakpointWidth(QUERIES.DESKTOP, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("justify-content", "flex-end", {
-      media: getBreakpointWidth(QUERIES.DESKTOP, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("margin-bottom", defaultTheme.orbit.spaceXSmall, {
-      media: getBreakpointWidth(QUERIES.DESKTOP, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("margin", "0 0 12px 0!important", {
-      media: getBreakpointWidth(QUERIES.DESKTOP, defaultTheme),
-      modifier: "& > *",
-    });
+
+  it.each(Object.entries(DIRECTIONS))("should have direction %s", (name, direction) => {
+    render(
+      <Stack flex direction={direction} dataTest="test">
+        <div>kek</div>
+        <div>bur</div>
+      </Stack>,
+    );
+
+    expect(screen.getByTestId("test")).toHaveStyle({ flexDirection: direction });
   });
-});
 
-describe("Stack with mobile and some desktop properties", () => {
-  const inline = true;
-  const direction = DIRECTIONS.ROW;
-  const wrap = true;
-  const grow = true;
-  const shrink = true;
-  const basis = "auto";
-  const align = ALIGNS.END;
-  const justify = JUSTIFY.END;
-  const spaceAfter = SPACINGS_AFTER.LARGE;
-  const spacing = SPACINGS.XXLARGE;
-  const desktop = {
-    shrink: false,
-    direction: DIRECTIONS.COLUMN,
-    spacing: SPACINGS.XLARGE,
-    basis: theme => `calc(25% - ${theme.orbit.spaceLarge})`,
-  };
+  it.each(Object.entries(JUSTIFY))("should change justify %s", (name, justify) => {
+    render(
+      <Stack flex justify={justify} dataTest="test">
+        <div>kek</div>
+        <div>bur</div>
+      </Stack>,
+    );
+
+    const getValue = (val: "start" | "end" | "between" | "center" | "around") => {
+      if (val === "around" || val === "between") return `space-${val}`;
+      if (val === "center") return "center";
+      return `flex-${val}`;
+    };
```

Comment:
To avoid logic like this, you can pass pairs of prop value and what result you expected to `it.each`:

```js
it.each([
  ["start", "flex-start"],
  ["end", "flex-end"],
  ["between", "space-between"],
  ["center", "center"],
  ["around", "space-around"],
])("should change justify %s", (justify, expected) => {
  // ...
})
```